File: Answers/02.validate-sequence/ValidateSequence.ts
/**
 *
 *
 */
function isValidSubsequence(array: number[], sequence: number[]) {
  let seqIndex = 0;
  for (const value of array) {
    if (sequence[seqIndex] === value) {
      seqIndex++
    }
  }

  return seqIndex === sequence.length
}

isValidSubsequence([5, 1, 22, 25, 6, -1, 8, 10], [1, 6, -1, 10]) //?

File: Answers/02.validate-sequence/ValidateSequence.go
package main

func IsValidSubsequence(array []int, sequence []int) bool {
	// Write your code here.
	seqIndex := 0

	for _, value := range array {
		if (seqIndex == len(sequence)) {
			break
		}
		if sequence[index] === value {
			seqIndex += 1
		}
	}

	return seqIndex == len(sequence)
}
